app/src/main/java/com/celzero/bravedns/net/go/GoVpnAdapter.java

```diff
@@ -285,9 +282,19 @@ public void checkForCryptRemoval() {
         }
     }
 
-    public void setCryptMode() {
+    public void setDnscryptMode(AppMode.TunnelMode tunnelMode) {
+        ExecutorService executor = Executors.newSingleThreadExecutor();
+        executor.execute(new Runnable() {
```

Comment:
Directly using instead of assignment as below. 
```
Executors.newSingleThreadExecutor().execute(() -> {
   if (isDoh(dnsMode) || isDnscrypt(dnsMode)) {
      setBraveDNSLocalMode();
      setBraveDNSRemoteMode(dohURL);
   }
});
```

Reply:
Changed to ktx.coroutines.